pkg/kubelet/volumemanager/volume_manager.go

```diff
@@ -481,6 +489,23 @@ func (vm *volumeManager) WaitForUnmount(ctx context.Context, pod *v1.Pod) error
 	return nil
 }
 
+func (vm *volumeManager) WaitForAllPodsUnmount(ctx context.Context, pods []*v1.Pod) error {
+	var errors []error
+	var wg sync.WaitGroup
+	wg.Add(len(pods))
+	for _, pod := range pods {
+		go func(pod *v1.Pod) {
+			defer wg.Done()
+			if err := vm.WaitForUnmount(ctx, pod); err != nil {
```

Comment:
The podAttachAndMountTimeout in WaitForUnmount is set to 2*time.Minute + 3*time.Second by default, perhaps we should expose this internal constant in the group.ShutdownGracePeriodSeconds configuration to let administrators know what is the optimal timeout value?

Reply:
Superseeded by https://github.com/kubernetes/kubernetes/pull/125070